lib/fli/FliImpl.h

```diff
@@ -227,14 +346,45 @@ class FliTimerCache {
     FliImpl *impl;
 };
 
+class FliIterator : public GpiIterator {
+public:
+    enum OneToMany {
+        OTM_END = 0,
+        OTM_CONSTANTS,  // include Generics
+        OTM_SIGNALS,
+        OTM_REGIONS,
+        OTM_SIGNAL_SUB_ELEMENTS,
+        OTM_VARIABLE_SUB_ELEMENTS
+    };
+
+    FliIterator(GpiImplInterface *impl, GpiObjHdl *hdl);
+
+    virtual ~FliIterator() { };
+
+    Status next_handle(std::string &name, GpiObjHdl **hdl, void **raw_hdl);
+
+private:
+    void populate_handle_list(OneToMany childType);
+
+private:
+    static GpiIteratorMapping<int, OneToMany> iterate_over;      /* Possible mappings */
+    std::vector<OneToMany> *selected;                            /* Mapping currently in use */
+    std::vector<OneToMany>::iterator one2many;
+
+    std::vector<void *> m_vars;
```

Comment:
Can a concrete type be used here rather than void?
And do the handles supplied via populate_handle_list need to be released/freed when the iterator is destroyed?